merkledag/utils/diffenum.go

```diff
@@ -65,27 +65,35 @@ type diffpair struct {
 // getLinkDiff returns a changeset between nodes 'a' and 'b'. Currently does
 // not log deletions as our usecase doesnt call for this.
 func getLinkDiff(a, b node.Node) []diffpair {
-	have := make(map[string]*node.Link)
-	names := make(map[string]*node.Link)
```

Comment:
Looks right. The `Name` property of links is a protonode thing (that needs to die).